packages/web-components/fast-foundation/src/combobox/combobox.ts

```diff
@@ -0,0 +1,563 @@
+
+        this.position = this.forcedPosition
+            ? this.positionAttribute
+            : currentBox.top > availableBottom
+            ? SelectPosition.above
+            : SelectPosition.below;
+
+        this.positionAttribute = this.forcedPosition
+            ? this.positionAttribute
+            : this.position;
+
+        this.maxHeight =
+            this.position === SelectPosition.above ? ~~currentBox.top : ~~availableBottom;
+    }
+
+    /**
+     * Synchronize the form-associated proxy and update the value property of the element.
+     *
+     * @param prev - the previous collection of slotted option elements
+     * @param next - the next collection of slotted option elements
+     *
+     * @internal
+     */
+    public slottedOptionsChanged(prev, next): void {
+        super.slottedOptionsChanged(prev, next);
+        this.updateValue();
+    }
+
+    /**
+     * @internal
+     */
+    public typeaheadBufferChanged(prev: string = "", next: string) {
+        const matchNext = next.toLowerCase();
+        this.filteredOptions = this._options.filter((o: ListboxOption) => {
+            return o.text.toLowerCase().startsWith(matchNext);
+        });
+
+        if (!this.filteredOptions.length) {
+            this.options = this._options;
+        }
+
+        const selectedIndex = this.isAutocompleteInline
+            ? this.options.findIndex(o => o.isSameNode(this.filteredOptions[0]))
+            : this.isAutocompleteList &&
+              this.filteredOptions.length !== this._options.length
+            ? 0
+            : -1;
+
+        if (this.isAutocompleteList) {
+            this._options.forEach(o => {
+                o.hidden = !this.filteredOptions.includes(o);
+            });
+        }
+
+        if (selectedIndex !== this.selectedIndex) {
+            this.selectedIndex = selectedIndex;
+        } else {
+            this.setSelectedOptions();
+        }
+
+        if (
+            next.length > prev.length &&
+            (this.isAutocompleteInline || this.isAutocompleteBoth)
+        ) {
+            this.setInlineSelection(this.typeaheadBuffer.length);
+        }
+    }
+
+    /**
+     * @internal
+     */
+    private updateValue(shouldEmit?: boolean) {
+        if (this.$fastController.isConnected) {
+            if (this.firstSelectedOption) {
+                this.value = this.firstSelectedOption.value;
+            } else {
+                this.value = this.control.value;
+            }
+        }
+
+        if (shouldEmit) {
+            this.$emit("change");
+        }
+    }
+}
+
+/**
+ * Includes ARIA states and properties relating to the ARIA combobox role.
+ *
+ * @public
+ */
+export class DelegatesARIACombobox extends ARIAGlobalStatesAndProperties {
+    /**
+     * See {@link https://www.w3.org/WAI/PF/aria/roles#button} for more information
+     * @public
+     * @remarks
+     * HTML Attribute: aria-expanded
+     */
+    @observable
```

Comment:
should this be attr?

Reply:
I'm actually not sure if this should even be presented in this class, since it's only used on the internal control and is handled programmatically.

Reply:
Is this set on the combobox itself or a control in the template?

Reply:
@chrisdholt it's set on the text input control within the template (it's similarly set on the button control for Select).

I think this property is also set on the `DelegatesARIAListbox` class which doesn't seem right. I'll readdress the ARIA properties on these three components.

Reply:
@radium-v sounds good - in the case that the listbox *supports* that role but doesn't need it by default, I think it's okay for it to be there, but agree that it likely shouldn't be there if the host element itself needs it as part of the public facing API.